app/models/user.rb

```diff
@@ -38,6 +38,7 @@ class User < ActiveRecord::Base
   has_many :permissions,       :through => :filters
   has_many :cached_usergroup_members
   has_many :widgets, :dependent => :destroy
+  has_many :ssh_keys, :dependent => :destroy
```

Comment:
It would be great if you defined a Jail class and allow this for safe mode so so we could do something like `@host.owner.ssh_keys` in provisioning and job templates. Also it would be probably handy to have a Jail definition in SshKey class as well so we could print the public key and potentially fingerprint. Another bonus points for macro that would print public keys for a given user. No push, just an idea :-)